### PR TITLE
Work on both Python 2 and Python 3

### PR DESCRIPTION
**Pull request by pkgw**
Let's see if we can get the test suite passing on Python 3.

## Superseded:

At the moment, no work done here except to turn on the Python 3 CI.

The errors are dominaeted by a mode that looks pretty innocuous; it looks like there are cases where Python is starting to complain about files not being explicitly closed, which makes the tests that count warnings fail:

```
FAIL: Test writing data with compression filters
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/a/lib/python3.6/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/a/hera/pyuvdata/pyuvdata/tests/test_uvh5.py", line 127, in test_UVH5CompressionOptions
    uvtest.checkWarnings(uv_in.read_uvfits, [uvfits_file], message='Telescope EVLA is not')
  File "/a/hera/pyuvdata/pyuvdata/tests/__init__.py", line 101, in checkWarnings
    assert(False)
AssertionError: 
-------------------- >> begin captured stdout << ---------------------
wrong number of warnings. Expected number was 1, actual number was 2.
warning 0 is: {message : UserWarning('Telescope EVLA is not in known_telescopes.',), category : 'UserWarning', filename : '/a/hera/pyuvdata/pyuvdata/uvfits.py', lineno : 443, line : None}
warning 1 is: {message : ResourceWarning("unclosed file <_io.FileIO name='/a/hera/pyuvdata/pyuvdata/data/day2_TDEM0003_10s_norx_1src_1spw.uvfits' mode='rb' closefd=True>",), category : 'ResourceWarning', filename : '/a/hera/pyuvdata/pyuvdata/uvdata.py', lineno : 1480, line : None}
```

### CODE REVIEW

**Comment by pkgw** [comment]
@bhazelton @plaplant The test suite now passes on my machine on both Python 2 and Python 3. Ready for review if the CI agrees!

**Comment by pkgw** [comment]
Oh, I forgot to look at the doctests.

I have to run off, but folks are more than welcome to push this over the finish line!

**Comment by pkgw** [comment]
Ready for review.

File: pyuvdata/beamfits.py [hunk 1]
```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
...
-        n_efield_dims = max([ax_nums[key] for key in ax_nums])
-
-        if self.beam_type == 'power':
-            # check for case where the data is complex (e.g. for xy beams)
-            if n_dimensions > ax_nums['complex'] - 1:
-                complex_arrs = np.split(data, 2, axis=0)
-                self.data_array = np.squeeze(complex_arrs[0] + 1j * complex_arrs[1], axis=0)
+            self.beam_type = primary_header.pop('BTYPE', None)
+            if self.beam_type is not None:
+                self.beam_type = self.beam_type.lower()
```
Comment (dannyjacobs) [inline pyuvdata/beamfits.py]: Where did the rest of this function go?

File: pyuvdata/beamfits.py [hunk 2]
```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
...
-            if primary_header.pop('CTYPE' + str(ax_nums['feed_pol'])).lower().strip() == 'feedind':
-                self.Nfeeds = primary_header.pop('NAXIS' + str(ax_nums['feed_pol']))
-            feedlist = primary_header.pop('FEEDLIST', None)
-            if feedlist is not None:
-                self.feed_array = np.array(feedlist[1:-1].split(', '))
-        else:
-            raise ValueError('Unknown beam_type: {type}, beam_type should be '
-                             '"efield" or "power".'.format(type=self.beam_type))
+            if self.beam_type == 'intensity':
+                self.beam_type = 'power'
```
Comment (dannyjacobs) [inline pyuvdata/beamfits.py]: This particular diff is kind of baffling.  Huge gaps in the HEAD suggesting large deletions. The bits which are lined up don't match.

File: pyuvdata/beamfits.py [hunk 3]
```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
...
+                self.Naxes1 = primary_header.pop('NAXIS' + str(ax_nums['img_ax1']))
+                self.Naxes2 = primary_header.pop('NAXIS' + str(ax_nums['img_ax2']))
+
+                self.axis1_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['img_ax1'])
+                self.axis2_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['img_ax2'])
 
                 # if units aren't defined they are degrees by FITS convention
                 # convert degrees to radians because UVBeam uses radians
-                axis1_units = basisvec_header.pop('CUNIT' + str(basisvec_ax_nums['img_ax1']), 'deg')
+                axis1_units = primary_header.pop('CUNIT' + str(ax_nums['img_ax1']), 'deg')
```
Comment (dannyjacobs) [inline pyuvdata/beamfits.py]: Why did 223 conversion require all these variable name changes?  If they're passing the tests, I suppose its fine. But why?

File: pyuvdata/beamfits.py [hunk 4]
```diff
@@ -50,276 +50,276 @@ def read_beamfits(self, filename, run_check=True, check_extra=True,
...
+                if primary_header.pop('CTYPE' + str(ax_nums['spw'])).lower().strip() == 'if':
+                    self.Nspws = primary_header.pop('NAXIS' + str(ax_nums['spw']), None)
+                    # subtract 1 to be zero-indexed
+                    self.spw_array = uvutils.fits_gethduaxis(primary_hdu, ax_nums['spw']) - 1
+
+            if n_dimensions > ax_nums['basisvec'] - 1:
+                if primary_header.pop('CTYPE' + str(ax_nums['basisvec'])).lower().strip() == 'vecind':
+                    self.Naxes_vec = primary_header.pop('NAXIS' + str(ax_nums['basisvec']), None)
+
+            if (self.Nspws is None or self.Naxes_vec is None) and self.beam_type == 'power':
```
Comment (dannyjacobs) [inline pyuvdata/beamfits.py]: Aha, heres the rest of the bits which were missing above.  This diff is hopeless. Lets just assume everythings fine.

File: pyuvdata/beamfits.py (hunk 3 above)
Comment (pkgw) [reply pyuvdata/beamfits.py]: I think what happened here is that the code that highlights the individual works got confused by the large diff chunks and showed something misleading. If you look at the diff with whitespace changes filtered out, you'll see that there aren't any variable renames happening here.

**Pull request merged**